### ISSUE TITLE
[Discord] Impossible to log in for prolonged periods of time via Miranda (clients+site works)

### ISSUE THREAD

**Issue by LittleVulpix** [miranda-ng/miranda-ng#755]
The protocol logs in, stays logged in for a second and then logs out.

Using the website (discord.gg) or the client on any of the available devices works, so this is something related to Miranda.

```
[23:18:13 0EA8] [DisCordless] Gateway connection gracefully closed
[23:18:13 0EA8] [DisCordless] CDiscordProto::ShutdownSession
[23:18:13 0EA8] [DisCordless] CDiscordProto::OnLoggedOut
[23:18:14 1168] [DisCordless] (04F94B30:1952) Connection closed internal
[23:18:14 1168] [DisCordless] (04F94B30:4294967295) Connection closed
[23:18:14 1168] [DisCordless] CDiscordProto::WorkerThread: leaving

```

**Comment by ziceptor** [comment]
Confirm.

Miranda NG Version: 0.95.6 build #18512 (5a7c41f) x64
Discord.dll v.0.5.0.3 [5 Oct 2017 23:28:38] - Discord protocol

**Comment by dartraiden** [comment]
Successfully logged today.

**Comment by LittleVulpix** [comment]
Yeah this is not like a "everyone has issues logging in" problem. It's more like.. something happens on your account or whatever and no matter what you try you just can't log in. I don't remember but there is some server switching which Miranda is not able to handle properly, so depending on your location, number of requests and "luck", if you get switched, it seems you get kicked off (in miranda). The web discord works fine however.

**Comment by Corak667** [comment]
Still not fixed:
[Discord] CDiscordProto::SetStatus iNewStatus = 40072, m_iStatus = 40071, 
m_iDesiredStatus = 40072 m_hWorkerThread = 00000260

Cannot login. Only miranda restart helps

**Comment by LittleVulpix** [comment]
That suggests another issue altogether. Are you using KeepStatus, perhaps? I've seen it bug out Discord this way before. Entire netlog would probably help (with core logging)

**Comment by Corak667** [comment]
Yes, I now logging core+discord libs and KeepStatus enabled... Will send netlog there, when eternal disconnect will manifest)

**Comment by Corak667** [comment]
eternal disconnect:
"
[13:14:15 01AC] [Discord.Gateway] (019161B0:324) Data sent
00000000: 81 8F 34 65-00 FE 4F 47-6F 8E 16 5F-31 D2 16 01  ЃЏ4e.юOGoЋ._1Т..
00000010: 22 C4 07 52-7D                                   "Д.R}
[13:14:15 0D30] [Discord] Got packet: buffer = 38, opcode = 1, headerSize = 2, final = 1, masked = 0
[13:14:15 0D30] [Discord] JSON received:
{"t":null,"s":null,"op":11,"d":null}
[13:14:29 0D30] [Discord] Got packet: buffer = 147, opcode = 1, headerSize = 4, final = 1, masked = 0
[13:14:29 0D30] [Discord] JSON received:
{"t":"PRESENCE_UPDATE","s":38,"op":0,"d":{"user":{"id":"103507602440216576"},"status":"offline","last_modified":1537179260039,"activities":[]}}
[13:14:29 0D30] [Discord] got a server command to dispatch: PRESENCE_UPDATE
[13:14:56 01AC] [Discord.Gateway] (019161B0:324) Data sent
00000000: 81 8F 0C D7-0F 64 77 F5-60 14 2E ED-3E 48 2E B3  ЃЏ.Ч.dwх`..н>H.і
00000010: 2D 5E 3F EF-72                                   -^?пr
[13:15:11 0D30] [Discord] Got packet: buffer = 4, opcode = 8, headerSize = 2, final = 1, masked = 0
[13:15:11 0D30] [Discord] server required to exit
[13:15:11 0D30] [Discord.Gateway] (019161B0:324) Connection closed internal
[13:15:11 0D30] [Discord.Gateway] (019161B0:4294967295) Connection closed
[13:15:11 0D30] [Discord] CDiscordProto::ShutdownSession
[13:15:11 0D30] [Discord] CDiscordProto::OnLoggedOut
[13:15:11 01AC] KeepStatus: connection lost! (Discord)
[13:15:12 0D24] [Discord] (0192E770:288) Connection closed internal
[13:15:12 0D24] [Discord] (0192E770:4294967295) Connection closed
[13:15:12 0D24] [Discord] CDiscordProto::WorkerThread: leaving
[13:15:21 01AC] KeepStatus: status for Discord differs: stored = 40072, real = 40071
[13:15:21 01AC] KeepStatus: assigning status 40072 (40072, 40072) to Discord
[13:15:21 01AC] [Discord] CDiscordProto::SetStatus iNewStatus = 40072, m_iStatus = 40071, m_iDesiredStatus = 40071 m_hWorkerThread = 00000000
[13:15:21 0CCC] [Discord] CDiscordProto::WorkerThread: entering
[13:15:21 0CCC] [Discord] Executing request #10:
https://discordapp.com/api/v6/users/@me
[13:15:21 0CCC] [Discord] Connection request to discordapp.com:443 (Flags 11)....
[13:15:21 0CCC] [Discord] (019161B0) Connecting to server discordapp.com:443....
[13:15:21 0CCC] [Discord] (019161B0) Connecting to ip 104.16.58.5:443 ....
[13:15:31 01AC] KeepStatus: status for Discord differs: stored = 40072, real = 1
[13:15:31 01AC] KeepStatus: Setting Discord offline before making a new connection attempt
[13:15:31 01AC] [Discord] CDiscordProto::SetStatus iNewStatus = 40071, m_iStatus = 1, m_iDesiredStatus = 40072 m_hWorkerThread = 00000128
[13:15:31 01AC] KeepStatus: assigning status 40072 (40072, 40072) to Discord
[13:15:31 01AC] [Discord] CDiscordProto::SetStatus iNewStatus = 40072, m_iStatus = 40071, m_iDesiredStatus = 40071 m_hWorkerThread = 00000128
[13:15:41 01AC] KeepStatus: status for Discord differs: stored = 40072, real = 40071
[13:15:41 01AC] KeepStatus: assigning status 40072 (40072, 40072) to Discord
[13:15:41 01AC] [Discord] CDiscordProto::SetStatus iNewStatus = 40072, m_iStatus = 40071, m_iDesiredStatus = 40072 m_hWorkerThread = 00000128
[13:15:42 0CCC] [Discord] (019161B0) Connecting to ip 104.16.59.5:443 ....
[13:15:51 01AC] KeepStatus: status for Discord differs: stored = 40072, real = 40071
[13:15:51 01AC] KeepStatus: assigning status 40072 (40072, 40072) to Discord
[13:15:51 01AC] [Discord] CDiscordProto::SetStatus iNewStatus = 40072, m_iStatus = 40071, m_iDesiredStatus = 40072 m_hWorkerThread = 00000128
[13:15:51 0CCC] [Discord] w:\miranda-build\src\mir_app\src\netlibopenconn.cpp 718: connect() failed (1460)
[13:15:51 0CCC] [Discord] Request 10 failed
[13:16:01 01AC] KeepStatus: status for Discord differs: stored = 40072, real = 40071
[13:16:01 01AC] KeepStatus: assigning status 40072 (40072, 40072) to Discord
[13:16:01 01AC] [Discord] CDiscordProto::SetStatus iNewStatus = 40072, m_iStatus = 40071, m_iDesiredStatus = 40072 m_hWorkerThread = 00000128
[13:16:11 01AC] KeepStatus: status for Discord differs: stored = 40072, real = 40071
[13:16:11 01AC] KeepStatus: assigning status 40072 (40072, 40072) to Discord
[13:16:11 01AC] [Discord] CDiscordProto::SetStatus iNewStatus = 40072, m_iStatus = 40071, m_iDesiredStatus = 40072 m_hWorkerThread = 00000128
[13:16:21 01AC] KeepStatus: status for Discord differs: stored = 40072, real = 40071
[13:16:21 01AC] KeepStatus: assigning status 40072 (40072, 40072) to Discord
[13:16:21 01AC] [Discord] CDiscordProto::SetStatus iNewStatus = 40072, m_iStatus = 40071, m_iDesiredStatus = 40072 m_hWorkerThread = 00000128
[13:16:31 01AC] KeepStatus: status for Discord differs: stored = 40072, real = 40071
[13:16:31 01AC] KeepStatus: assigning status 40072 (40072, 40072) to Discord
"
http://coraksoft1.narod.ru/temp/miranda/netlog_discord-eternal-disconnect.7z
